### PR TITLE
feat(typescript): Add typings for the module

### PR DESCRIPTION
**Pull request by wolfy1339**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  privateKey: string
+  baseUrl?: string
+  cache?: LRU.Cache<string, string>
+}
+declare interface getJWTOptions {
+  installation_id: string
+}
+
+declare interface getInstallationAccessTokenOptions {
+  installation_id: string
```
Comment (gr2m) [inline index.d.ts]:
```suggestion
  installation_id: number
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+// Not really a class, but it is how they say it should be used in the readme.
+// In TypeScript, you cannot use the `new` keyword on functions (excluding old-style classes using functions and prototype), only on classes
+declare class App {
+  constructor(options: AppOptions)
+  getSignedJsonWebToken(options?: getJWTOptions): string
```
Comment (gr2m) [inline index.d.ts]: `getSignedJsonWebToken` has no options.

```suggestion
  getSignedJsonWebToken(): string
```

Comment (gr2m) [reply index.d.ts]: and should the methods be prefixed with `public `? Compare https://github.com/probot/probot/blob/e1d78738aa4791940a5161aed2364e42dbc1ce6d/src/%40types/%40octokit/app/index.d.ts

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import LRU = require("lru-cache");
```
Comment (gr2m) [inline index.d.ts]: should that be `import LRU from 'lru-cache'`?

Comment (wolfy1339) [reply index.d.ts]: That's the TypeScript equivalent (it transpires down to) to CommonJS import syntax

**Pull request merged**